pytorch_lightning/trainer/trainer.py

```diff
@@ -765,6 +852,10 @@ def fit(self, model):
         elif self.single_gpu:
             self.single_gpu_train(model)
 
+        elif self.use_tpu:
+            log.info(f'training on {self.num_tpu_cores} TPU cores')
+            xmp.spawn(self.tpu_train, args=(model,), nprocs=self.num_tpu_cores, start_method='fork')
```

Comment:
This [Colab from the `pytorch/xla` documentation](https://github.com/pytorch/xla) suggest to use the `spawn` method instead of `fork` in `xmp.spawn()` -- `fork` seems to be more unstable and resource intensive than `spawn`, see [documentation here](https://docs.python.org/3/library/multiprocessing.html). 

They also say, however, that `fork` is the only supported method in Colab. So maybe Trainer would need to figure out if it is running in a Colab environment and call the right method? Something like this could work:

```suggestion
            #  COLAB_GPU is an env var available by default in Colab environments.
            start_method = 'fork' if os.getenv('COLAB_GPU') else 'spawn'
            xmp.spawn(self.tpu_train, args=(model,), nprocs=self.num_tpu_cores, start_method=start_method)
```